docs/dep/dep-0001.rst

```diff
@@ -0,0 +1,599 @@
+Start with an idea for DebOps
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The DEP process begins with a new idea for DebOps. It is highly
+recommended that a single DEP contain a single key proposal or new
+idea. Small enhancements or patches often don't need
+a DEP and can be injected into the DebOps development workflow with a
+patch submission to the `DebOps issue tracker`_. The more focused the
+DEP, the more successful it tends to be. The DEP editors reserve the
+right to reject DEP proposals if they appear too unfocused or too
+broad. If in doubt, split your DEP into several well-focused ones.
+
+Each DEP must have a champion -- someone who writes the DEP using the style and
+format described below, shepherds the discussions in the appropriate forums,
+and attempts to build community consensus around the idea. The DEP champion
+(a.k.a. Author) should first attempt to ascertain whether the idea is DEP-able.
+Posting to the <debops-users@lists.debops.org> mailing list is the best way to
+go about this.
+
+Vetting an idea publicly before going as far as writing a DEP is meant
+to save the potential author time. Many ideas have been brought
+forward for changing DebOps that have been rejected for various
+reasons [2]_. Asking the DebOps community first if an idea is original
+helps prevent too much time being spent on something that is
+guaranteed to be rejected based on prior discussions (searching
+the Internet does not always do the trick). It also helps to make sure
+the idea is applicable to the entire community and not just the author.
+Just because an idea sounds good to the author does not
+mean it will work for most people in most areas where DebOps is used.
+
+Once the champion has asked the DebOps community as to whether an idea has any
+chance of acceptance, a draft DEP should be presented to the DebOps mialing
+list. This gives the author a chance to flesh out the draft DEP to make
+properly formatted, of high quality, and to address initial concerns about the
+proposal.
+
+
+Submitting a DEP
+~~~~~~~~~~~~~~~~
+
+Following a discussion on the DebOps mailing list, the proposal should be
+submitted as a draft DEP via a `GitHub pull request`_. The draft must be
+written in DEP style as described below, else it will fail review immediately
+(although minor errors may be corrected by the editors).
+
+The standard DEP workflow is:
+
+- You, the DEP author, fork the `DebOps monorepo`_, and create a file named
+  :file:`docs/dep/dep-9999.rst` that contains your new DEP. Use "9999" as your
+  draft DEP number.
+- Push this to your GitHub fork and submit a pull request.
+- The DEP editors review your PR for structure, formatting, and other errors.
+- Once approved, they will assign your DEP a number, and label it as Standards
+  Track, Informational, or Process, and give it the status "Draft"
+
+Once the review process is complete, and the DEP editors approve it (note that
+this is *not* the same as accepting your DEP!), they will squash commit your
```

Comment:
Good point. In Python as I understand it, this was done to keep the `git` repository with the PEPs clean. Original author of the PEP could have done multiple commits against PEP 9999, and after passing it off to the PEP editors they were free to renumber the PEP to the correct new number and squash all of the author commits together.

But we are tracking the commit origins via GPG signatures which breaks this system... I suppose that some of the current PEP workflow would need to be adapted for DEPs to handle this. I'll add a quick explanation on how to squash commits before submitting the work so that DEP authors could still have their work sighed with GPG keys, and there would need to be a mechanism to request new DEP numbers. I guess a reservation list in DEP 0 would suffice and DEP authors could be given the numbers after discussing an idea over the mailing list or GitHub issues.